app/src/main/java/org/simple/clinic/summary/PatientSummaryUpdate.kt

```diff
@@ -99,20 +99,39 @@ class PatientSummaryUpdate : Update<PatientSummaryModel, PatientSummaryEvent, Pa
   }
 
   private fun contactDoctorClicked(model: PatientSummaryModel): Next<PatientSummaryModel, PatientSummaryEffect> {
-    return when (model.teleconsultInfo) {
-      // TODO (SM): Check phone numbers before dispatching
-      //  LoadPatientTeleconsultationInfo to show contact doctor sheet
-      is TeleconsultInfo.Fetched -> dispatch(LoadPatientTeleconsultationInfo(
+    return when (val teleconsultInfo = model.teleconsultInfo) {
+      is TeleconsultInfo.Fetched -> {
+        val effect = eventForContactDoctorClicked(teleconsultInfo, model)
+        dispatch(effect)
+      }
+      else -> noChange()
+    }
+  }
+
+  private fun eventForContactDoctorClicked(
+      teleconsultInfo: TeleconsultInfo.Fetched,
+      model: PatientSummaryModel
+  ): PatientSummaryEffect {
+    return if (areMultipleDoctorsAvailable(teleconsultInfo)) {
```

Comment:
I don't really think this needs to be a function since it is only used in one place. Maybe we can convert it to a computed property on the `TeleconsultInfo.Fetched` class instead?

Reply:
Ah right, forgot about that. Done